tutorials/3d/3d_rendering_limitations.rst

```diff
@@ -0,0 +1,125 @@
+.. _doc_3d_rendering_limitations:
+
+3D rendering limitations
+========================
+
+Introduction
+------------
+
+Due to their focus on performance, real-time rendering engines have many
+limitations. Godot renderer's is no exception. To work effectively with those
+limitations, you need to understand them.
+
+Texture size limits
+-------------------
+
+Modern desktop/laptop GPUs can only render textures up to 16384×16384. If you
+need to display a larger texture for some reason, you'll need to split it into
+multiple smaller textures. Old GPUs like the GeForce 8/9 series are limited to
+8192×8192.
+
+Mobile GPUs are typically limited to 4096×4096 textures. Therefore, if you want
+your texture to display correctly on all platforms, you should avoid using
+textures larger than 4096×4096.
+
```

Comment:
I think *very* recent GPUs like the NVIDIA RTX line may actually up max texture resolution to `32768x32768`. 

Also `Old GPUs like the GeForce 8/9 series are limited to 8192×8192` may be misleading from what I've found?
Geforce 9 would be from 2008. I can't find specs for those, but I can't imagine them supporting anything above `4096x4096`, if that.

NVIDIA Geforce 900, i.e. GTX 950 to 980Ti/TitanX from 2014 still seems to support only up to `4096x4096`.
The Geforce 10 line (2016), i.e. GTX 1030 to 1080 Ti finally seems to have upped this to `16384x16384`.

GPUs around the Geforce 9xx line and below seem to still be fairly popular according to the current [Steam hardware survey data](https://store.steampowered.com/hwsurvey/Steam-Hardware-Software-Survey-Welcome-to-Steam), maybe they even make up the majority (at least for non powerusers).

So maybe we should point out that you want to stay at or below `4096x4096` if you possibly want to support GPUs older than 3 years. (Which is probably the case for the majority of games, besides the next Crysis.)

Also maybe add a note about preferring power-of-two texture sizes for various reasons (like mipmapping/compression, and sometimes speed? Not an expert.), and about adding padding around textures to prevent some problems with texture bleeding.

Reply:
I had a GeForce 9600M back in the day. I remember seeing its maximum texture size being set to 8192x8192 according to OpenGL.

Either way, I should make a test project and send it to people with different hardware configurations to confirm this.

Reply:
Yeah, you might be right :)

It seems very hard to find exact specs online, so I could only search for things like `"GTX 950" "Maximum Texture Dimension Size"`, which yields to results like [these](https://github.com/strands-project/strands_3d_mapping/issues/103):
>Device 0: "GeForce GTX 950"
>...
>Maximum Texture Dimension Size (x,y,z) 1D=(65536), 2D=(65536, 65536), 3D=(4096, 4096, 4096)

Or [these](http://cuda-memo.blogspot.com/2015/03/geforce-gtx-970-devicequery.html):
>Device 0: "GeForce GTX 970"
>...
>Maximum Texture Dimension Size (x,y,z)         1D=(65536), 2D=(65536, 65536), 3D=(4096, 4096, 4096)

Reply:
Older phones can be limited to 2048x2048 textures. Also on mobile, fill rate and texture bandwidth can be key as it uses lots of power so they do everything to reduce this. So using low res textures is highly recommended where possible. Also consider using mipmap nearest instead of trilinear filtering. And texture compression presumably helps loads too.